### DISCUSSION TITLE
bash: /root/bin/pip: Permission denied

### DISCUSSION THREAD

**Discussion by YepXbruce** [apache/airflow#37974] [General]
### Apache Airflow version

2.8.2

### What happened?

when i run docker . 
`docker -D -l debug run  --entrypoint ""   apache/airflow:2.8.2-python3.10 pip list`
then logs error from the console.

bash: /root/bin/pip: Permission denied
DEBU[0000] [hijack] End of stdout       

### How to reproduce

`docker -D -l debug run  --entrypoint ""   apache/airflow:2.8.2-python3.10 pip list`

### Operating System

```
PRETTY_NAME="Ubuntu 23.04"
NAME="Ubuntu"
VERSION_ID="23.04"
VERSION="23.04 (Lunar Lobster)"
VERSION_CODENAME=lunar
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=lunar
LOGO=ubuntu-logo
```

### Deployment

Other

### Deployment details

```
Client: Docker Engine - Community
 Cloud integration: v1.0.35+desktop.11
 Version:           25.0.2
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        29cf629
 Built:             Thu Feb  1 00:23:03 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          25.0.2
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       fce6e0c
  Built:            Thu Feb  1 00:23:03 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
```
Client: Docker Engine - Community
 Cloud integration: v1.0.35+desktop.11
 Version:           25.0.2
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        29cf629
 Built:             Thu Feb  1 00:23:03 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          25.0.2
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       fce6e0c
  Built:            Thu Feb  1 00:23:03 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
Use instead `docker run -it --rm apache/airflow:2.8.2-python3.10 python -m pip list`

**Comment by potiuk** [comment]
Could you please explain WHY did you expec that putting empty entrypoint and expected it to work and even opened an issue about it - indicating that this is an Airflow issue @YepXbruce . I am really puzzled what made you come to the conclusion, which documentation piece you looked at led you to this and why you thought it was a bug? I would love to understand your line of though here?

Since you already took some time of ours to look at your issue - can you please at least explain the reasoning? I think we deserve it - and it might help others who will see that issue and your explanation to understand what they do wrong if they arrive to the same conclusion.

Can you help us with that?
